### PR TITLE
Extends matrix-js-sdk types with RawEvent definition

### PR DESCRIPTION
**Pull request by Tbaut**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/matrix-js-sdk/index.d.ts [hunk 1]
```diff
@@ -683,12 +684,12 @@ export class Filter {
...
-  forwardLooking: boolean;      //  True if this event is 'forward looking', meaning that getDirectionalContent() will return event.content and not event.prev_content.
+  event: RawEvent;         //  The raw (possibly encrypted) event. Do not access this property directly unless you absolutely have to. Prefer the getter methods defined
+  sender: RoomMember;      //  The room member who sent this event, or null e.g. this is a presence event. This is only guaranteed to be set for events that appear in
+  target: RoomMember;      //  The room member who is the target of this event, e.g. the invitee, the person being banned, etc.
+  status: EventStatus;     //  The sending status of the event.
+  error: Error;            //  most recent error associated with sending the event, if any
+  forwardLooking: boolean; //  True if this event is 'forward looking', meaning that getDirectionalContent() will return event.content and not event.prev_content.
 
   constructor(event: object)
   getType(): EventType;
```
Comment (andrevmatos) [inline types/matrix-js-sdk/index.d.ts:695]: Can you please also add the `getContent` getter, since you're at it?

Comment (Tbaut) [reply types/matrix-js-sdk/index.d.ts:695]: wow, that was quick, sure, I'll add the `getType` and `getContent` :)

Comment (Tbaut) [reply types/matrix-js-sdk/index.d.ts]: done, along with some other getters based on https://github.com/matrix-org/matrix-js-sdk/blob/fb8976167190e81e4139da8501068c3bd0523d82/src/models/event.js#L185-L345

**Review by andrevmatos — APPROVED** [review]
lgtm, thanks

**Pull request merged**